### PR TITLE
fix/filter-refactor

### PR DESCRIPTION
**Pull request by ItayBenMoshe**
### This PR implements the following changes:

-  move SearchBar to common folder
-  use SearchBar in investigationTable and usersManagement
- change the flow of the filter process in investigationTable
- make the Badge inside the filter icon appear only when the user filter something
- fix bug [1309](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1309/)

### CODE REVIEW

File: client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts [hunk 1]
```diff
@@ -1,56 +1,52 @@
...
+        {
+            search: {
                 or: [
                     { userName: { includes: value } },
                     { id: { includes: value } }
                 ]
-            } : { or: null };
-        }
+            }
+        } : { search: null };
```
Comment (Noam6421) [inline client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts:11]: syntax, maybe better:
{  option1  }
:
{  option2  }

File: client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts [hunk 2]
```diff
@@ -1,56 +1,52 @@
...
-                sourceOrganizationBySourceOrganization: { displayName: { in: values } }
-            } : { sourceOrganizationBySourceOrganization: null };
+    SOURCE_ORGANIZATION: (values: string[]) => {
+        return values.length > 0 ?
+        { 
+            sourceOrganization: {
+                sourceOrganizationBySourceOrganization: { displayName: { in: values } } 
+            } 
         }
+        : { sourceOrganization: null };
```
Comment (Noam6421) [inline client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts:20]: same

File: client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts [hunk 3]
```diff
@@ -1,56 +1,52 @@
...
-        create(values: string[]) {
-            return values.length > 0 ? 
-            {
-                userType: { in: values }
-            } : { userType: null };
-        }
+    USER_TYPE: (values: string[]) => {
+        return values.length > 0 ? 
+        { userType: { userType: { in: values } } }
+        : { userType: null };
```
Comment (Noam6421) [inline client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts:43]: same

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 4]
```diff
@@ -31,9 +31,11 @@ interface CellNameSort {
 
 const usersManagementTitle = 'ניהול משתמשים';
 const sourceOrganizationLabel = 'מסגרת';
+const searchBarLabel = 'הכנס שם או שם משתמש...';
+
 const notActiveSortFields: string[] = [UsersManagementTableHeadersNames.WATCH, UsersManagementTableHeadersNames.LANGUAGES,
-UsersManagementTableHeadersNames.COUNTY, UsersManagementTableHeadersNames.USER_TYPE,
-UsersManagementTableHeadersNames.DESK];
+                                       UsersManagementTableHeadersNames.COUNTY, UsersManagementTableHeadersNames.USER_TYPE,
+                                       UsersManagementTableHeadersNames.DESK];
```
Comment (Noam6421) [inline client/src/components/App/Content/UsersManagement/UsersManagement.tsx:38]: if the spacing is long, fix it... not sure if it's git visualization

**Review by Noam6421 — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/FilterCreators.ts [hunk 5]
```diff
@@ -1,110 +1,107 @@
...
+        if (timeRangeFilter.id !== allTimeRangeId) {
+            return { [InvestigationsFilterByFields.TIME_RANGE]: { 
+                creationDate: { 
+                    greaterThanOrEqualTo: timeRangeFilter.startDate,
+                    lessThanOrEqualTo: timeRangeFilter.endDate } 
+                }
+            } ;
         } else {
-            return {}
+            return { [InvestigationsFilterByFields.TIME_RANGE]: null }
```
Comment (y-golde) [inline client/src/components/App/Content/LandingPage/InvestigationTable/FilterCreators.ts:102]: isn't is supposed to be { [InvestigationsFilterByFields.TIME_RANGE]: {} ?

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 6]
```diff
@@ -162,42 +163,69 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
     const axiosInterceptorId = useSelector<StoreStateType, number>(state => state.investigation.axiosInterceptorId);
     const windowTabsBroadcastChannel = useRef(new BroadcastChannel(BC_TABS_NAME));
 
+    useEffect(() => {
+        const statusFilterToSet = statusFilter.length > 0 ? filterCreators.STATUS(statusFilter) : null;
+        const deskFilterToSet = deskFilter.length > 0 ? filterCreators.DESK_ID(deskFilter) : null;
+        const timeRangeFilterToSet = timeRangeFilter.id !== 10 ? filterCreators.TIME_RANGE(timeRangeFilter) : null;
```
Comment (y-golde) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:169]: maybe export 10 to const?

Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:169]: like INVALID_WHATEVER_STATUS_CODE

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 7]
```diff
@@ -162,42 +163,69 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
...
+        const unAssignedFilterToSet = (unassignedUserFilter && !inactiveUserFilter) ? filterCreators.UNASSIGNED_USER(unassignedUserFilter) : null;
+        const inActiveToSet = (inactiveUserFilter && !unassignedUserFilter) ? filterCreators.INACTIVE_USER(unassignedUserFilter) : null;
+        const unAllocatedToSet = (unassignedUserFilter && unassignedUserFilter) ? filterCreators.UNALLOCATED_USER(unassignedUserFilter) : null;
```
Comment (y-golde) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:172]: maybe change to just 2 ifs  it looks like they are very dependent

**Review by y-golde — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts (hunk 6 above)
Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:169]: `allTimeRangeId`

File: client/src/components/App/Content/LandingPage/InvestigationTable/FilterCreators.ts [hunk 8]
```diff
@@ -1,110 +1,107 @@
...
-        if (timeRangeFilter.id === 10) {
-            return {};
-        } else if (timeRangeFilter) {
-            return { creationDate: { greaterThanOrEqualTo: timeRangeFilter.startDate, lessThanOrEqualTo: timeRangeFilter.endDate }} 
+        if (timeRangeFilter.id !== allTimeRangeId) {
+            return { [InvestigationsFilterByFields.TIME_RANGE]: { 
+                creationDate: { 
+                    greaterThanOrEqualTo: timeRangeFilter.startDate,
+                    lessThanOrEqualTo: timeRangeFilter.endDate } 
+                }
```
[the remark this answers is not in the document]
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/FilterCreators.ts]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/FilterCreators.ts [hunk 9]
```diff
@@ -1,110 +1,107 @@
...
     [InvestigationsFilterByFields.TIME_RANGE]: (timeRangeFilter: TimeRange) => {
-        if (timeRangeFilter.id === 10) {
-            return {};
-        } else if (timeRangeFilter) {
-            return { creationDate: { greaterThanOrEqualTo: timeRangeFilter.startDate, lessThanOrEqualTo: timeRangeFilter.endDate }} 
+        if (timeRangeFilter.id !== allTimeRangeId) {
+            return { [InvestigationsFilterByFields.TIME_RANGE]: { 
+                creationDate: { 
+                    greaterThanOrEqualTo: timeRangeFilter.startDate,
+                    lessThanOrEqualTo: timeRangeFilter.endDate }
```
[the remark this answers is not in the document]
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/FilterCreators.ts]: fixed

File: client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts (hunk 1 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts]: fixed

File: client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts (hunk 2 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts]: fixed

File: client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts (hunk 3 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersFilter/FilterCreators.ts]: fixed

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx (hunk 4 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/UsersManagement/UsersManagement.tsx:38]: fixed

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts (hunk 6 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts]: fixed

**Review by Noam6421 — APPROVED** [review]

**Review by y-golde — APPROVED** [review]

**Pull request merged**